contracts/token/ERC1155/ERC1155.sol

```diff
@@ -376,10 +379,26 @@ contract ERC1155 is Context, ERC165, IERC1155, IERC1155MetadataURI {
         }
     }
 
-    function _asSingletonArray(uint256 element) private pure returns (uint256[] memory) {
-        uint256[] memory array = new uint256[](1);
-        array[0] = element;
+    function _asSingletonArrays(
+        uint256 element1,
+        uint256 element2
+    ) private pure returns (uint256[] memory, uint256[] memory) {
+        uint256[] memory array1;
+        uint256[] memory array2;
```

Comment:
It might be better to use the already declared return variables

```suggestion
    ) private pure returns (uint256[] memory array1, uint256[] memory array2) {
```

We won't need the return at the end